ExampleMod/Content/Projectiles/ExamplePaperAirplaneProjectile.cs

```diff
@@ -0,0 +1,166 @@
+				Projectile.ai[0] = -Main.rand.Next(30, 80); // Set Projectile.ai[0] to a random number from -30 to -79.
+				Projectile.netUpdate = true; // Sync the projectile in a multiplayer game.
+			}
+
+			// Kill the projectile if it touches a liquid. (It will automatically get killed by touching a tile. You can change that by returning false in OnTileCollide())
+			if (Projectile.wet && Projectile.owner == Main.myPlayer) {
+				Projectile.Kill();
+			}
+
+			Projectile.ai[0] += 1f; // Increase Projectile.ai[0] by 1 every tick. Remember, there are 60 ticks per second.
+
+			Vector2 rotationVector = Projectile.rotation.ToRotationVector2() * 8f; // Get the rotation of the projectile.
+
+			float ySinModifier = (float)Math.Sin((float)Math.PI * 2f * (float)(Main.timeForVisualEffects % 90.0 / 90.0)) * Projectile.direction * Main.WindForVisuals; // This will make the projectile fly in a sine wave fashion.
+
+			Vector2 newVelocity = rotationVector + new Vector2(Main.WindForVisuals, ySinModifier); // Create a new velocity using the rotation and wind.
+			_ = Vector2.UnitX * Projectile.direction; // Set the sign of the x unit to the same sign as the direction.
+
+			bool directionSameAsWind = Projectile.direction == Math.Sign(Main.WindForVisuals) && Projectile.velocity.Length() > 3f; // True if the projectile is moving the same direction as the wind and is not moving slowly.
+			bool readyForFlip = Projectile.ai[0] >= 20f && Projectile.ai[0] <= 69f; // True if Projectile.ai[0] is between 20 and 69
+
+			// Once Projectile.ai[0] reaches 70...
+			if (Projectile.ai[0] == 70f) {
+				Projectile.ai[0] = -Main.rand.Next(120, 600); // Set it back to a random number from -120 to -599.
+			}
+
+			// Do a flip! This will cause the projectile to fly in a loop if directionSameAsWind and readyForFlip are true.
+			if (readyForFlip && directionSameAsWind) {
+				float lerpValue = Utils.GetLerpValue(0f, 30f, Projectile.ai[0], clamped: true);
+				newVelocity = rotationVector.RotatedBy((-Projectile.direction) * ((float)Math.PI * 2f) * 0.02f * lerpValue);
+			}
+
+			Projectile.velocity = newVelocity.SafeNormalize(Vector2.UnitY) * Projectile.velocity.Length(); // Set the velocity to the value we calculated above.
+
+			// If it is flying normally. i.e. not flying a loop.
+			if (!(readyForFlip && directionSameAsWind)) {
+				float yModifier = MathHelper.Lerp(0.15f, 0.05f, Math.Abs(Main.WindForVisuals));
+
+				// Half of time, decrease the y velocity a little.
+				if (Projectile.timeLeft % 40 < 20) {
+					Projectile.velocity.Y -= yModifier;
+				}
+				// The other half of time, increase the y velocity a little.
+				else {
+					Projectile.velocity.Y += yModifier;
+				}
+
+				// Cap the y velocity so the projectile falls slowly and doesn't rise too quickly
+				if (Projectile.velocity.Y < -2f) {
+					Projectile.velocity.Y = -2f;
+				}
+				if (Projectile.velocity.Y > 2f) {
+					Projectile.velocity.Y = 2f;
+				}
```

Comment:
```suggestion
				// Cap the y velocity so the projectile falls slowly and doesn't rise too quickly
				Projectile.velocity.Y = Math.Clamp(Projectile.velocity.Y, -2f, 2f);
```

Clamp is already used below, so why not here